### ISSUE TITLE
Move `PrecalculatedFeaturesAudioSource` inside `OnlineSpeakerDiarization`

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/StreamingSpeakerDiarization#84]
## Problem

In retrospective, this implementation is conceptually wrong. An audio source shouldn't emit features. The class is too coupled to `OnlineSpeakerDiarization`.

This is a problem because it makes the API clunky and not very flexible for other applications.

## Idea

A better design for pre-computed features would be to use RxPY operators to accumulate `batch_size` chunks, then apply segmentation and embedding models, and finally flatmap into an observable of segmentation and embedding chunks so that they're given to the online clustering block in order.
An additional advantage is that this could even run in a true streaming fashion because batches will be created on-the-fly. For example, it could allow users to fine-tune a pipeline on a remote dataset that doesn't fit in their disk or that they can't download permanently because of privacy concerns.

## Example

With the current reactive pipeline we need to modify `OnlineSpeakerDiarization.from_audio_source()` this can be managed by a `batch_size` parameter that could be set to 1 in cases where we want real time streaming inference:

```python
batch_size = 32
source = MicrophoneAudioSource(16000)
segmentation = SegmentationModel.from_pyannote("pyannote/segmentation")
embedding = EmbeddingModel.from_pyannote("pyannote/embedding")
source.stream.pipe(
    dops.regularize_audio_stream(...),
    ops.buffer_with_count(count=batch_size),
    ops.map(lambda chunks: (
        chunks, torch.stack([c.data for c in chunks]),
    )),
    ops.starmap(lambda chunks, batch: (
        chunks, segmentation(batch), embedding(batch)
    )),
    ops.flat_map(lambda batch: rx.from_iterable(zip(*batch))),
    # Online clustering + aggregation goes here
    ...
).subscribe(lambda pred_wav: print(*pred_wav))

source.read()
```

With an rx-free pipeline, the following would be user code adapting only inputs and outputs with their favorite streaming library (in this example RxPY):

```python
batch_size = 32
source = MicrophoneAudioSource(16000)
diarization = OnlineSpeakerDiarization()
source.stream.pipe(
    dops.regularize_audio_stream(...),
    ops.buffer_with_count(count=batch_size),
    ops.map(diarization),
    ops.flat_map(lambda results: rx.from_iterable(*results)),
).subscribe(lambda pred_wav: print(*pred_wav))

source.read()
```

If `OnlineSpeakerDiarization` receives a `SlidingWindowFeature`, it runs in real time. If it receives a list of `SlidingWindowFeature`, it runs batched inference.

In this scenario, `OnlineSpeakerDiarization` doesn't need to know about the streaming library.
